board/raspberrypi/overlay/usr/bin/streameye.sh

```diff
@@ -143,70 +43,100 @@ function invalid_opt() {
-            fi
-            if [ -n "${height}" ]; then
-                video_extras="${video_extras},height=${height}"
+            valid_opts=("analoggain" "awb" "awbgains" "bitrate" "brightness" "colfx" "contrast" "denoise" "digitalgain" "drc" "ev" "exposure" "flicker" "framerate" "height" "hflip" "imxfx" "intra" "irefresh" "level" "metering" "profile" "roi" "rotation" "saturation" "sharpness" "shutter" "vflip" "vstab" "width" "mjpegbitrate" "mjpegframerate" "mjpegwidth" "mjpegheight")
+            raspimjpeg_opts="--videoout ${video_path}"
+            while read line; do
+                key=$(echo ${line} | cut -d ' ' -f 1)
+                if invalid_opt "${key}" "${valid_opts[@]}"; then
+                    continue
+                fi
+                if echo "${line}" | grep false &>/dev/null; then
+                    continue
+                fi
+                if echo "${line}" | grep true &>/dev/null; then
+                    line=${key}
+                fi
+                raspimjpeg_opts="${raspimjpeg_opts} --${line}"
+            done < ${RASPIMJPEG_CONF}
+
+            video_iso=$(grep -e ^iso ${RASPIMJPEG_CONF} | cut -d ' ' -f 2)
+            if [ -n "${video_iso}" ]; then
+                raspimjpeg_opts="${raspimjpeg_opts} --ISO ${video_iso}"
             fi
-            if [ -n "${framerate}" ]; then
-                video_extras="${video_extras},framerate=${framerate}/1"
-            fi
-            video_opts="v4l2src device=${video_path} ! video/x-h264${video_extras} ! h264parse ! rtph264pay name=pay0 pt=96"
 
             if [ -r ${MOTIONEYE_CONF} ] && grep 'log-level debug' ${MOTIONEYE_CONF} >/dev/null; then
+                raspimjpeg_opts="${raspimjpeg_opts} -v"
                 streameye_opts="${streameye_opts} -d"
             fi
 
-            test-launch -p ${RTSP_PORT} -m h264 "\"( ${video_opts} ${audio_opts} )\"" &>${GSTREAMER_LOG} &
-            sleep 10
-            gst-launch-1.0 -v rtspsrc location=rtsp://127.0.0.1:${RTSP_PORT}/h264 latency=0 drop-on-latency=1 ! rtph264depay ! h264parse ! omxh264dec ! videorate ! video/x-raw,framerate=5/1 ! jpegenc ! filesink location=/dev/stdout | streameye ${streameye_opts} &>${STREAMEYE_LOG} &
-            sleep 5
-        fi
+            rtspserver_opts="${rtspserver_opts} -P ${RTSP_PORT} -u h264"
+            video_framerate=$(grep -e ^framerate ${RASPIMJPEG_CONF} | cut -d ' ' -f 2)
+            if [ -n "${video_framerate}" ]; then
+                rtspserver_opts="${rtspserver_opts} -F ${video_framerate}"
+            fi
 
-        iptables -D INPUT -p tcp --dport ${RTSP_PORT} -j DROP
-        iptables -D INPUT -p tcp -s localhost --dport ${RTSP_PORT} -j ACCEPT
+            mjpeg_opts="--mjpegbitrate ${MJPEG_BITRATE} --mjpegwidth ${MJPEG_WIDTH} --mjpegheight ${MJPEG_HEIGHT} --mjpegframerate ${MJPEG_FRAMERATE}"
+            let "GOP=$((video_framerate))*2"
+            raspimjpeg_opts="${raspimjpeg_opts} --intra $GOP ${mjpeg_opts}"
```

Comment:
We should probably check whether `intra` is specified in `raspimjpeg.conf` file before setting a default of `framerate*2`.